test/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotAddOrRemoveStaticKeywordTests.cs

```diff
@@ -0,0 +1,282 @@
+                CreateType(" static class", " public static void F() {}"),
+                // We expect one difference for the type becoming static and one for the method
+                // becoming static (since static class requires static members)
+                CreateDifferences((DifferenceType.Added, DiagnosticIds.CannotAddStaticToType, "T:CompatTests.First"),
+                                  (DifferenceType.Added, DiagnosticIds.CannotAddStaticToMember, "M:CompatTests.First.F")),
+            };
+        }
+
+        public static IEnumerable<object[]> RemoveStaticFromTypeCases()
+        {
+            // Remove static from class - this is compatible
+            yield return new object[] {
+                CreateType(" static class", " public static void F() {}"),
+                CreateType(" class", " public static void F() {}"),
+                CreateDifferences(), // No differences expected
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(RemoveStaticCases))]
+        [MemberData(nameof(AddStaticCases))]
+        [MemberData(nameof(AddStaticToTypeCases))]
+        [MemberData(nameof(RemoveStaticFromTypeCases))]
+        public static void EnsureDiagnosticIsReported(string leftSyntax, string rightSyntax, CompatDifference[] expected)
+        {
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new(s_ruleFactory);
+
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right });
+
+            Assert.Equal(expected, differences);
+        }
+
+        [Fact]
+        public static void EnsureNoCrashWhenMembersDoNotExist()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First {
+    public static void F() {}
+  }
+}
+";
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First {
+    public static void G() {}
+  }
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            // Register CannotAddOrRemoveStaticKeyword and MemberMustExist rules as this test validates both.
+            ApiComparer differ = new(s_ruleFactory.WithRule((settings, context) => new MembersMustExist(settings, context)));
+
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(left, right);
+
+            CompatDifference[] expected = new[]
+            {
+                CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.MemberMustExist, string.Empty, DifferenceType.Removed, "M:CompatTests.First.F"),
+            };
+            Assert.Equal(expected, differences);
+        }
+
+        [Fact]
+        public static void ReproductionCaseFromIssue()
```

Comment:
We can omit this case.  It's covered by the others, and there's nothing special about this, I just created it as an illustrative minimal repro.